docs/design/mui-best-practices.md

```diff
@@ -0,0 +1,44 @@
+---
+title: MUI Best Practices
+description: Guidelines for using MUI to maximise compatibility with Docker Desktop
+keywords: Docker, extensions, mui, theme, theming, material-ui, material
+---
+
+# MUI Best Practices
+
+This article assumes you're following our recommended practice by using our [MUI theme](https://www.npmjs.com/package/@docker/docker-mui-theme). Following the steps below will maximise compatibility with Docker Desktop and minimise the work you need to do as an extension author. They should be considered supplementary to the non-MUI-specific guidelines found in the [UI kit article](ui-kit.md).
+
+## Assume the theme can change at any time
+
+It’s tempting to fine-tune your UI with precise colors, offsets and font sizings to make it look as attractive as possible. Please resist that temptation. Any specialisations you make today will be relative to the current MUI theme, and likely will look worse when the theme changes. Any part of the theme might change without warning, including (but not limited to):
+
+-  The font, or font sizes
+-  Border thicknesses or styles
+-  Colors, e.g.:
+   -  Our palette members (e.g. `red-100`) could change their RGB values
+   -  The semantic colors (e.g. `error`, `primary`, `textPrimary`, etc) could be changed to use a different member of our palette
+   -  Background colors (e.g. those of the page, or of dialogs) could change
+-  Spacings, e.g.:
+   -  The size of the basic unit of spacing (exposed via `theme.spacing`) - for instance, we could allow users to customise the density of the UI
+   -  The default spacing between paragraphs or grid items
+
+The best way to build your UI, so that it’s robust against future theming changes, is to:
+
+-  Override the default styling as little as possible.
+-  Prefer semantic typography. e.g. use `Typography`s or `Link`s with appropriate `variant`s instead of using typographical HTML elements (`<a>`, `<p>`, `<h1>`, etc) directly.
```

Comment:
```suggestion
-  Use semantic typography. e.g. use `Typography`s or `Link`s with appropriate `variant`s instead of using typographical HTML elements (`<a>`, `<p>`, `<h1>`, etc) directly.
```